lib/epiviewpoint/bulk_fhir_parser.ex

```diff
@@ -0,0 +1,205 @@
+         extension: extensions
+       }) do
+    %{
+      caseid: caseid,
+      person_tid: person_tid,
+      search_firstname: search_firstname,
+      search_lastname: search_lastname,
+      dateofbirth: format_date(dateofbirth),
+      sex: String.capitalize(to_string(sex)),
+      diagaddress_street1: diagaddress_street1,
+      diagaddress_city: diagaddress_city,
+      diagaddress_state: diagaddress_state,
+      diagaddress_zip: diagaddress_zip,
+      phonenumber: phonenumber,
+      ethnicity: find_extension(extensions, "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity"),
+      occupation: find_extension(extensions),
+      race: find_extension(extensions, "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race")
+    }
+  end
+
+  defp extract_observation(%EpiViewpoint.R4.Observation{
+         id: lab_result_tid,
+         subject: %EpiViewpoint.R4.Reference{reference: "Patient/" <> pat_id},
+         effective_date_time: datecollected,
+         issued: resultdate,
+         code: %EpiViewpoint.R4.CodeableConcept{text: testname},
+         interpretation: [%EpiViewpoint.R4.CodeableConcept{coding: [%EpiViewpoint.R4.Coding{display: result}]}],
+         performer: [%EpiViewpoint.R4.Reference{reference: "Organization/" <> org_id}],
+         extension: extensions
+       }) do
+    %{
+      lab_result_tid: lab_result_tid,
+      pat_id: pat_id,
+      datecollected: datecollected,
+      resultdate: format_date(resultdate),
+      testname: testname,
+      result: result,
+      org_id: org_id,
+      datereportedtolhd: find_extension(extensions)
+    }
+  end
+
+  defp extract_organization(%EpiViewpoint.R4.Organization{
+         id: organization_id,
+         name: ordering_facility_name
+       }) do
+    %{
+      organization_id: organization_id,
+      ordering_facility_name: ordering_facility_name
+    }
+  end
+
+  defp find_extension(extensions, url \\ nil) do
+    Enum.find_value(extensions, fn
+      %EpiViewpoint.R4.Extension{
+        url: ^url,
+        extension: [%EpiViewpoint.R4.Extension{url: "ombCategory", value_coding: %EpiViewpoint.R4.Coding{display: value}}, _]
+      } ->
+        value
+
+      %EpiViewpoint.R4.Extension{url: "http://hl7.org/fhir/StructureDefinition/patient-occupation", value_string: value} ->
+        value
+
+      %EpiViewpoint.R4.Extension{url: "http://hl7.org/fhir/StructureDefinition/datereportedtolhd", value_date: value} ->
+        value
+
+      _ ->
+        nil
+    end)
+  end
+
+  defp join_resources(resources) do
+    patients = Map.get(resources, "Patient", [])
+    observations = Map.get(resources, "Observation", [])
+    organizations = Map.get(resources, "Organization", [])
+
+    joined = Enum.map(observations, &merge_resources(&1, patients, organizations))
+
+    {:ok, joined}
+  end
+
+  defp merge_resources(observation, patients, organizations) do
+    patient = Enum.find(patients, &(&1.caseid == observation.pat_id))
```

Comment:
I think it's cleaner to use the `default` for `Enum.find(patients, %{}, &(&1...)` where the 2nd arg is the default than to `|| %{}` in the `Map.merge/2` functions later.

Reply:
yes, thanks!